http_server/src/main.rs

```diff
@@ -0,0 +1,246 @@
+              include_str!("../../credentials/cert.pem"),
+              include_str!("../../credentials/issuer.pem"),
+        ]).unwrap());
+        worker
+    };
+}
+
+async fn req_to_vec_body(request: Request<Body>) -> Result<Request<Vec<u8>>> {
+    let (parts, body) = request.into_parts();
+    let body = hyper::body::to_bytes(body).await?.to_vec();
+    Ok(Request::from_parts(parts, body))
+}
+
+async fn resp_to_vec_body(response: Response<Body>) -> Result<Response<Vec<u8>>> {
+    let (parts, body) = response.into_parts();
+    let body = hyper::body::to_bytes(body).await?.to_vec();
+    Ok(Response::from_parts(parts, body))
+}
+
+struct HttpsFetcher<'a>(&'a HttpsClient);
+
+#[async_trait]
+impl Fetcher for HttpsFetcher<'_> {
+    async fn fetch(&self, request: HttpRequest) -> Result<HttpResponse> {
+        let request: Request<Vec<u8>> = request.try_into()?;
+        let request: Request<Body> = request.map(|b| b.into());
+
+        let response: Response<Body> = self.0.request(request).await?;
+        // TODO: Do something streaming.
+        resp_to_vec_body(response).await?.try_into()
+    }
+}
+
+async fn generate_sxg_response(
+    fallback_url: &Uri,
+    payload: Response<Body>,
+) -> Result<Response<Body>> {
+    let payload: HttpResponse = resp_to_vec_body(payload).await?.try_into()?;
+    let cert_origin = format!(
+        "{}://{}",
+        fallback_url
+            .scheme()
+            .ok_or_else(|| anyhow!("fallback url missing scheme"))?,
+        fallback_url
+            .authority()
+            .ok_or_else(|| anyhow!("fallback url missing authority"))?
+    );
+    let subresource_fetcher = HttpsFetcher(&HTTPS_CLIENT);
+    let runtime = sxg_rs::runtime::Runtime {
+        now: std::time::SystemTime::now(),
+        sxg_signer: Box::new(WORKER.create_rust_signer()?),
+        fetcher: Box::new(subresource_fetcher),
+        ..Default::default()
+    };
+    let sxg = WORKER
+        .create_signed_exchange(
+            &runtime,
+            sxg_rs::CreateSignedExchangeParams {
+                payload_body: &payload.body,
+                payload_headers: WORKER.transform_payload_headers(payload.headers)?,
+                skip_process_link: false,
+                status_code: 200,
+                fallback_url: &format!("{}", fallback_url),
+                cert_origin: &cert_origin,
+                // The fastly crate provides only read access to dictionaries, so
+                // header integrities cannot be cached. However, I believe the
+                // subresource_fetcher will go through the cache.
+                header_integrity_cache: sxg_rs::http_cache::NullCache {},
+            },
+        )
+        .await?;
+    let sxg: Response<Vec<u8>> = sxg.try_into()?;
+    Ok(sxg.map(Body::from))
+}
+
+async fn serve_preset_content(url: &str) -> Option<PresetContent> {
+    let ocsp_fetcher = HttpsFetcher(&HTTPS_CLIENT);
+    // TODO: Store OCSP in database?
```

Comment:
We don't need this because`WORKER.serve_preset_content` internally reads and writes OCSP in `runtime.storage`.

Reply:
Clarified the TODO is about persistent storage.